amethyst_network/src/simulation/transport/tcp.rs

```diff
@@ -213,7 +215,10 @@ fn write_message(
     channel: &mut EventChannel<NetworkSimulationEvent>,
 ) {
     if let Some((_, stream)) = net.get_stream(message.destination) {
-        if let Err(e) = stream.write(&message.payload) {
+        let len: Bytes = Bytes::copy_from_slice(&(message.payload.len() as u32).to_le_bytes());
```

Comment:
Doesn't `u32` introduce too much overhead for small messages? Maybe not. But to solve this and the other issue I mentioned I think it would be better to use some existing encoding libraries instead of inventing stuff. Although, I don't have any experience with this to suggest something.